file_store/src/file_sink.rs

```diff
@@ -363,24 +360,28 @@ impl FileSink {
                     let prev_path = active_sink.path.clone();
                     self.deposit_sink(&prev_path).await?;
                     self.active_sink = Some(self.new_sink().await?);
+                } else {
+                    active_sink.transport.send(buf).await?;
+                    active_sink.size += buf_len;
                 }
             }
-            // No sink, make a new one
+            // No sink, make a new one and send the incoming buf
+            // if the write is not too large.
             None => {
-                self.active_sink = Some(self.new_sink().await?);
+                let mut new_sink = self.new_sink().await?;
+                assert!(new_sink.size == 0);
+                if buf_len < self.max_size {
+                    new_sink.transport.send(buf).await?;
+                    new_sink.size += buf_len;
+                    self.active_sink = Some(new_sink);
+                } else {
+                    // XXX: How can we directly deposit here since the incoming
+                    // buf is already too big?
```

Comment:
This indicates this is not correct also.. The whole point was to deposit a sink and make a new one if the new buffer will overrun the sink. Then write to the sink. 

I still don't get how you have an empty sink at all. I'm missing the logic error in the original implementation and this patch adds incorrect behavior